### ISSUE TITLE
strict-family checks (strictPropertyInitialization, useUnknownInCatchVariables) incorrectly enabled when only strictNullChecks is set — regression in dev.20260206.1

### ISSUE THREAD

**Issue by JerryPeace** [microsoft/typescript-go#4145]
### 🔎 Summary

Starting with `7.0.0-dev.20260206.1`, `tsgo` reports `TS2564` (`strictPropertyInitialization`) and `TS18046` (`useUnknownInCatchVariables`) for a project whose `tsconfig.json` enables **`strictNullChecks` only** — without `strict: true`.

These two checks belong to the `strict` family and, per `tsc` behavior, must **not** be enabled unless `strict: true` (or the individual flag) is set. `tsc` 5.9 does not report them for this config. This is a regression: `dev.20260205.1` matches `tsc`; `dev.20260206.1` does not.

### 🧩 Minimal repro

`tsconfig.json`:
```json
{
  "compilerOptions": {
    "strictNullChecks": true,
    "noImplicitAny": true,
    "target": "ES2021",
    "module": "commonjs"
  },
  "include": ["repro.ts"]
}
```

`repro.ts`:
```ts
class Foo {
  name: string;        // unexpected TS2564
}

function bar() {
  try {
    // ...
  } catch (e) {
    console.log(e.message);   // unexpected TS18046
  }
}

export { Foo, bar };
```

Run: `tsgo --noEmit -p tsconfig.json`

### ✅ Expected vs ❌ Actual

| Compiler | tsconfig | Result |
|---|---|---|
| `tsc` 5.9.3 | `strictNullChecks` only | ✅ 0 errors |
| `tsc` 5.9.3 | `strict: true` | ❌ TS2564 + TS18046 (correct) |
| `tsgo` dev.20260205.1 | `strictNullChecks` only | ✅ 0 errors |
| `tsgo` **dev.20260206.1** | `strictNullChecks` only | ❌ TS2564 + TS18046 (**wrong**) |
| `tsgo` dev.20260206.1 | `strict: true` | ❌ TS2564 + TS18046 (correct) |

Actual output on `dev.20260206.1` with the repro above:
```
repro.ts(4,3): error TS2564: Property 'name' has no initializer and is not definitely assigned in the constructor.
repro.ts(12,17): error TS18046: 'e' is of type 'unknown'.
```

### 📉 Regression boundary (bisected)

- Last good: `7.0.0-dev.20260205.1`
- First bad: `7.0.0-dev.20260206.1`

### 💥 Impact

Real-world impact on a NestJS backend (only `strictNullChecks` enabled, idiomatic class-validator DTOs without field initializers): the upgrade surfaced **2000+ false errors** across the codebase that `tsc` does not report. The bug only manifests when `strict` is **off** but `strictNullChecks` is **on**; projects with `strict: true` (e.g. typical Vite/React frontends) are unaffected because those checks are legitimately enabled there.

### 🌐 Environment

- `@typescript/native-preview` `7.0.0-dev.20260206.1` (first bad)
- Reference: `typescript` 5.9.3
- OS: macOS (darwin arm64)

**Comment by jakebailey** [comment]
You should be checking against TS 6.0, not 5.9.

https://devblogs.microsoft.com/typescript/announcing-typescript-6-0/#simple-default-changes
